script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,112 @@
+  newVersion: string | number;
+};
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const redirectsPath = path.resolve(__dirname, '../src/redirects/');
+
+const bookLoader = makeUnifiedBookLoader(
+  createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`),
+  createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`)
+);
+
+async function updateRedirections(_bookId: string, currentVersion: string, _newVersion: string) {
+  if (currentVersion === _newVersion) {
+    return 0;
+  }
+
+  const { tree: currentTree, slug: bookSlug } = await bookLoader(_bookId, currentVersion)
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${_bookId} with defaultVersion ${currentVersion}`);
+      throw error;
+    });
+
+  const { tree: newTree } = await bookLoader(_bookId, _newVersion)
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${_bookId} with newVersion ${_newVersion}`);
+      throw error;
+    });
+
+  const redirectsBookPath = path.resolve(redirectsPath, bookId + '.json');
+
+  try {
+    // Create file with empty array only if it does not exixts
+    fs.writeFileSync(redirectsBookPath, '[]', { encoding: 'utf8', flag: 'wx' });
+  } catch {
+    // This will throw if file exists
+  }
+
+  const redirects: Redirects = require(redirectsBookPath);
+
+  const flatCurrentTree = flattenArchiveTree(currentTree);
+
+  const findRedirect = (section: LinkedArchiveTreeNode) => (
+    { pageId, bookId: pageBookId, pathname }: Redirects[0]
+  ) => pageId === section.id && pageBookId === _bookId && pathname.split('/').pop() === section.slug;
+
+  const formatSection = (section: LinkedArchiveTreeNode) => ({
+    bookId: _bookId,
+    pageId: section.id,
+    pathname: content.getUrl({ book: { slug: bookSlug }, page: { slug: section.slug } }),
+  });
+
+  let countNewRedirections = 0;
+  for (const section of flatCurrentTree) {
+    const { slug } = findArchiveTreeNodeById(newTree, section.id) || {};
+    if (
+      (slug && slug !== section.slug)
+      && !redirects.find(findRedirect(section))
+    ) {
+      redirects.push(formatSection(section));
+      countNewRedirections++;
+    }
+  }
+
+  fs.writeFileSync(redirectsBookPath, JSON.stringify(redirects, undefined, 2) + '\n', 'utf8');
+
+  return countNewRedirections;
+}
+
+async function processBook() {
+  const { defaultVersion } = books[bookId] || {};
+
+  if (defaultVersion === newVersion.toString()) {
+    console.log(`${bookId} alredy at desired version.`); // tslint:disable-line:no-console
+    process.exit(0);
+  }
+
+  const { title, version } = await bookLoader(bookId, newVersion.toString())
+    .catch((error) => {
+      // tslint:disable-next-line: no-console
+      console.log(`error while loading book ${bookId} with version ${newVersion}`);
+      throw error;
+    });
+
+  books[bookId] = { defaultVersion: version };
```

Comment:
why does this need to be changed in the config?

Reply:
I think I don't understand  the question
it is adding or updating the `defaultVersion` in `books` variable which is then saved as `config.books.json`

Reply:
yea, i see now, that wasn't clear to me.

it would have been nice to have a function with a descriptive name and/or create a new variable rather than modifying the global config.

you might have done something like that i haven't looked at the new changes yet

Reply:
fixed